### PR TITLE
fix duration can not be paresd

### PR DESCRIPTION
**Pull request by theScrabi**
this will fix the duration error that came up recently:
https://github.com/TeamNewPipe/NewPipeExtractor/issues/176

### CODE REVIEW

**Comment by Stypox** [comment]
Oh, by skimming through YoutubeStreamExtractor I also found out why NewPipe always fetches location-dependent strings instead of the chosen ones. It is because `downloader.download(...)` is called only with the url and not with the localization parameter. But this is also for another pr.

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamExtractorAgeRestrictedTest.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public void testGetUploaderName() throws ParsingException {
     @Ignore // Currently there is no way get the length from restricted videos
     @Test
     public void testGetLength() throws ParsingException {
-        assertTrue(extractor.getLength() > 0);
+        assertEquals(0, extractor.getLength());
```
[the remark this answers is not in the document]
Comment (theScrabi) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamExtractorAgeRestrictedTest.java]: It needs to, as soon as the ignore is being removed.

**Comment by theScrabi** [comment]
This:
grafik

The F****** extractor is literally untestable at the moment. Everytime you run it other tests fail. We need to fix that otherwise our whole CI is B***S***.

**Comment by theScrabi** [comment]
Or there is a S***load of changes on the youtube webpage coming around, and sometimes the extractor hits those servers with the new site.

13 failed. This is a catastrophe.

**Comment by Stypox** [comment]
🤔 Many of the errors are caused by ReCaptcha...

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 2]
```diff
@@ -685,11 +702,7 @@ private EmbeddedInfo getEmbeddedInfo() throws ParsingException, ReCaptchaExcepti
                 playerUrl = HTTPS + playerUrl;
             }
 
-            // Get embed sts
-            final String stsPattern = "\"sts\"\\s*:\\s*(\\d+)";
-            final String sts = Parser.matchGroup1(stsPattern, embedPageContent);
-
-            return new EmbeddedInfo(playerUrl, sts);
+            return new EmbeddedInfo(playerUrl, "");
```
[the remark this answers is not in the document]
Comment (theScrabi) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java]: I see. Similar to youtube-dl, ill bring it back.

**Comment by theScrabi** [comment]
> thinking Many of the errors are caused by ReCaptcha...

uff. So how are we going to tell the the CI?

**Comment by Stypox** [comment]
What about creating [a custom test runner](https://www.logicbig.com/tutorials/unit-testing/junit/runner.html) that, for example, ignores ReCaptcha exceptions and only generates warnings for classes where more than 60% of test functions fail?
Screenshot_20190731_103406
This happened to me once, and I wasn't able to reproduce it afterwards. No ReCaptcha exception was thrown, but the downloaded page lacked content anyway. With the custom runner you would just be warned that YoutubePlaylistExtractorRest$HugePlaylist couldn't be run.

Another option would be to wait some time before every network request: this way youtube should not create robot-verification problems.

**Comment by theScrabi** [comment]
Yes :D, the idea with a custom test runner sounds good. Was it possible to "re run" the test if it failed? For that we could check in the setup function of the test class weather the request went through well.

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamExtractorAgeRestrictedTest.java [hunk 3]
```diff
@@ -71,10 +71,9 @@ public void testGetUploaderName() throws ParsingException {
...
     }
 
-    @Ignore // Currently there is no way get the length from restricted videos
     @Test
     public void testGetLength() throws ParsingException {
-        assertTrue(extractor.getLength() > 0);
+        assertEquals(1789, extractor.getLength());
     }
 
     @Test
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamExtractorAgeRestrictedTest.java]: Remove `@Ignore` from `testGetAudioStreams()`, now it seems to work.

Comment (theScrabi) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamExtractorAgeRestrictedTest.java]: All right. However now subtitle tests seem to fail on my end.

**Pull request merged**